### PR TITLE
Fix a segfault of rabin2 on rpi4

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

A segmentation fault I met when using Cutter on rpi4.

**Test plan**

Buy an rpi4 and build r2.
```sh
rabin2 -Vj /bin/bunzip2
```
Get a segmentation fault.
Apply this change and rebuild.
No segmentation fault (at least here).

**Closing issues**

...

### CODE REVIEW

**Comment by ret2libc** [comment]
Do you have any idea why the segfault? It's not exactly clear to me what this fixes a crash

**Comment by eagleoflqj** [comment]
Arm use register to pass arg. On rpi4 (32 bit), if the arg is 64bit, it uses 2 registers. Seeing a "%llu", the _IO_vfprintf_internal picks 2 registers, but the second register is for the second argument which should be parsed by "%s".

**Comment by ret2libc** [comment]
Cool now it makes sense! Thanks for the explanation!!

**Comment by radare** [comment]
better use the PJ api so this kind of problems dont appear when someone changes a type of a random struct

**Comment by eagleoflqj** [comment]
Now the code seems cleaner.

File: libr/core/cbin.c [hunk 1]
```diff
@@ -3649,52 +3649,57 @@ static void bin_pe_versioninfo(RCore *r, int mode) {
 }
 
 static void bin_elf_versioninfo(RCore *r, int mode) {
-	const char *format = "bin/cur/info/versioninfo/%s%d";
+	const char *const format = "bin/cur/info/versioninfo/%s%d";
```
Comment (radare) [inline libr/core/cbin.c:3652]: Double const is not needed

File: libr/core/cbin.c [hunk 2]
```diff
@@ -3649,52 +3649,57 @@ static void bin_pe_versioninfo(RCore *r, int mode) {
...
 	int num_versym;
 	int num_verneed = 0;
 	int num_version = 0;
 	Sdb *sdb = NULL;
 	const char *oValue = NULL;
-	bool firstit_for_versym = true;
+	PJ *pj;
 	if (IS_MODE_JSON (mode)) {
-		r_cons_printf ("{\"versym\":[");
+		pj = pj_new ();
```
Comment (radare) [inline libr/core/cbin.c:3660]: Check for null

File: libr/core/cbin.c [hunk 3]
```diff
@@ -3649,52 +3649,57 @@ static void bin_pe_versioninfo(RCore *r, int mode) {
...
+		pj_o (pj);
+		pj_ka (pj, "versym");
 	}
 	for (num_versym = 0;; num_versym++) {
-		const char *versym_path = sdb_fmt (format, "versym", num_versym);
+		const char *const versym_path = sdb_fmt (format, "versym", num_versym);
```
Comment (radare) [inline libr/core/cbin.c:3665]: Same

File: libr/core/cbin.c (hunk 2 above)
Comment (eagleoflqj) [reply libr/core/cbin.c:3660]: Simply returns if null, because other calls to pj_new either don't check or just do so.

**Pull request merged**